src/odemis/driver/test/pigcs_test.py

```diff
@@ -601,6 +602,48 @@ def setUp(self):
         self.kwargs = KWARGS_CL
         self.kwargs_two = KWARGS_TWO_CL
 
+    #    @skip("faster")
+    def test_reference(self):
+        """
+        Test referencing for 2 axes.
+        """
+        stage = CLASS(**self.kwargs_two)
+
+        # Test proper referencing
+        stage.reference({'x'}).result()
+        self.assertTrue(stage.referenced.value['x'])
+        ref_pos_x = stage.position.value["x"]  # Get the reference position
+
+        stage.reference({'y'}).result()
+        self.assertTrue(stage.referenced.value['y'])
+        ref_pos_y = stage.position.value["x"]  # Get the reference position
+
+        # move to some position different from ref pos by certain percentage of axes range
+        stage.moveAbs({'x': ref_pos_x + stage.axes["x"].range[1] * 0.1}).result()
+        stage.moveAbs({'y': ref_pos_y + stage.axes["y"].range[1] * 0.2}).result()  # in case ref pos is 0 only add
+        self.assertAlmostEqual(stage.position.value['x'], ref_pos_x + stage.axes["x"].range[1] * 0.1, places=3)
+        self.assertAlmostEqual(stage.position.value['y'], ref_pos_y + stage.axes["y"].range[1] * 0.2, places=3)
+
+        # reference again
+        stage.reference({'x'}).result()
+        stage.reference({'y'}).result()
+        self.assertTrue(stage.referenced.value['x'])
+        self.assertTrue(stage.referenced.value['y'])
+        self.assertAlmostEqual(stage.position.value["x"], ref_pos_x, places=3)
+        self.assertAlmostEqual(stage.position.value["y"], ref_pos_y, places=3)
```

Comment:
Please also check the referencing of multiple axes. Like: `stage.reference({"x", "y"})`.